src/hotspot/share/gc/shared/gcConfiguration.cpp

```diff
@@ -75,7 +80,12 @@ GCName GCConfiguration::old_collector() const {
   }
 
   if (UseShenandoahGC) {
-    return Shenandoah;
+#if INCLUDE_SHENANDOAHGC
+    if (strcmp(ShenandoahGCMode, "generational") == 0) {
+      return Shenandoah;
+    }
+#endif
+    return NA;
```

Comment:
You moved the order between Shenandoah and ZGC in `young_collector()`, so you should probably do the same here.

Reply:
Thanks.  We've made your suggested change.